test/webgpu_required_limits.cpp

```diff
@@ -0,0 +1,98 @@
+#include <cassert>
+#include <emscripten.h>
+#include <iostream>
+#include <webgpu/webgpu.h>
+
+int adapter_limit_maxColorAttachmentBytesPerSample = -1;
+
+EM_ASYNC_JS(int, get_limit, (), {
+  if (!navigator.gpu) {
+    throw Error("WebGPU not supported.");
+  }
+
+  const adapter = await navigator.gpu.requestAdapter();
+  if (!adapter) {
+    throw Error("Couldn't request WebGPU adapter.");
+  }
+
+  console.log("adapter limits", adapter.limits);
+
+  return adapter.limits.maxColorAttachmentBytesPerSample;
```

Comment:
It looks kind of like we are peaking past the C/C++ API here and directly accessing the webgpu JS API.. is that purely so that we can verify that the C/C++ API reporting the same thing as the underlying JS API?   Is that OK/normal/expected for webGPU testing?

Reply:
So we are trying to request a webGPU device with limits higher than the default limits and if we get a device, we want the device limits to fulfill the requested limits. For I set only `maxColorAttachmentBytesPerSample`, because this was previously ignored (i.e. requesting a device with `maxColorAttachmentBytesPerSample` did not fail but returned a device that only supported the default value for `maxColorAttachmentBytesPerSample`).

But we can only test requesting higher-than-default limits, if the physical device (webGPU adapter) actually supports higher-than-default limits. So, I first check if the supported limit is higher than the default one (which is currently 32 according to the [webGPU spec](https://www.w3.org/TR/webgpu/#dom-supported-limits-maxcolorattachmentbytespersample)) and if not, I let the test pass. The reason for using the JS API directly here is that the C API bindings are not implemented yet: 

https://github.com/emscripten-core/emscripten/blob/6c23c37f2575e7249164911e10cbc06df752a4dd/src/library_webgpu.js#L2539-L2542

I hope letting the test pass when we can't request the higher limit is an alright way to do it (although the test case passing isn't really meaningful then).

PS: Its my first time writing a test-case that is depending on HW limits, please be patient with me here 🥲

Reply:
Sounds reasonable to me, I just wanted to check my assumptions.

@kainino0x is the authority on this stuff so I'll leave lgtm them.

Reply:
Now that I think about it, implementing `wgpuAdapterGetLimits` is trivial (and the implementation would be pretty much identical to `wgpuDeviceGetLimits`. Maybe encapsulate filling the WGPULimit struct in a new function in `LibraryWebGPU` (if is this the right place to add such a function?).

I think it would make this test case slightly more useful (and eliminate the need for accessing the JS API as well).

Would I have to open a new PR or can I just add it as part of this one?

Reply:
Sure, let's go ahead and do that. It would resolve one small concern I have that this test currently relies on comparing two different adapter objects which aren't guaranteed to be the same.